src/daemon/rpmostreed-transaction-livefs.c

```diff
@@ -0,0 +1,688 @@
+                   "TARGET_COMMIT=%s", target_csum,
+                   NULL);
+  g_string_truncate (journal_msg, 0);
+
+  /* Ensure that we have a rollback deployment that matches our booted checksum,
+   * so that if something goes wrong, the user can get to it.  If we have an
+   * older rollback, that gets GC'd.
+   */
+  OstreeDeployment *rollback_deployment = get_rollback_deployment (sysroot, booted_deployment);
+  if (!rollback_deployment)
+    {
+      if (!prepare_rollback_deployment (sysroot, repo, booted_deployment, cancellable, error))
+        return g_prefix_error (error, "Preparing rollback: "), FALSE;
+    }
+
+  /* Reload this, the sysroot may have changed it */
+  booted_deployment = ostree_sysroot_get_booted_deployment (sysroot);
+
+  /* Load SELinux policy for making changes to /etc */
+  g_autoptr(OstreeSePolicy) sepolicy = ostree_sepolicy_new_at (deployment_dfd, cancellable, error);
+  if (!sepolicy)
+    return FALSE;
+
+  /* Crack open our booted root */
+  if (!ostree_sysroot_deployment_set_mutable (sysroot, booted_deployment, TRUE,
+                                              cancellable, error))
+    return g_prefix_error (error, "Setting deployment mutable: "), FALSE;
+  /* This xattr says which commit we're in the process of overlaying */
+  if (fsetxattr (deployment_dfd, RPMOSTREE_LIVE_INPROGRESS_XATTR, target_csum, OSTREE_SHA256_STRING_LEN, 0) < 0)
+    return glnx_throw_errno_prefix (error, "Setting %s", RPMOSTREE_LIVE_INPROGRESS_XATTR);
+
+  rpmostree_output_task_begin ("Overlaying /usr");
+
+  if (!checkout_add_usr (repo, deployment_dfd, diff, target_csum, cancellable, error))
+    return FALSE;
+
+  /* Start replacing the rpmdb. First, ensure the temporary dir for the new
+     version doesn't exist */
+  if (!glnx_shutil_rm_rf_at (deployment_dfd, orig_rpmdb_path, cancellable, error))
+    return FALSE;
+  /* Check out the new rpmdb */
+  { OstreeRepoCheckoutAtOptions rpmdb_checkout_opts = { .mode = OSTREE_REPO_CHECKOUT_MODE_NONE,
+                                                        .no_copy_fallback = TRUE,
+                                                        .subpath = "/usr/share/rpm" };
+
+
+    if (!ostree_repo_checkout_at (repo, &rpmdb_checkout_opts, deployment_dfd, orig_rpmdb_path,
+                                  target_csum, cancellable, error))
+      return FALSE;
+  }
+  /* Now, RENAME_EXCHANGE the two */
+  if (glnx_renameat2_exchange (deployment_dfd, "usr/share/rpm", deployment_dfd, orig_rpmdb_path) < 0)
+    return glnx_throw_errno_prefix (error, "%s", "rename(..., RENAME_EXCHANGE) for rpmdb");
+  /* And nuke the old one */
+  if (!glnx_shutil_rm_rf_at (deployment_dfd, orig_rpmdb_path, cancellable, error))
+    return FALSE;
+
+  if (requires_etc_merge)
+    {
+      if (!copy_new_config_files (repo, origin_merge_deployment,
+                                  deployment_dfd, sepolicy, diff,
+                                  cancellable, error))
+        return FALSE;
```

Comment:
Wait...what do you mean?  We're using `.overwrite_mode = OSTREE_REPO_CHECKOUT_OVERWRITE_ADD_FILES` so they shouldn't right?